view/js/addon-hooks.js

```diff
@@ -0,0 +1,10 @@
+var addon_hooks={};
+
+function Addon_registerHook( type, hookfnstr )
```

Comment:
What is `hookfnstr` supposed to be?

Can you please add a comment block to explain what the function does?